src/server/server_family.h

```diff
@@ -257,6 +256,9 @@ class ServerFamily {
 
   std::shared_ptr<LastSaveInfo> last_save_info_;  // protected by save_mu_;
   std::atomic_bool is_saving_{false};
+  // If a save operation is currently in progress, calling this function will provide information
+  // about the memory consumption during the save operation.
+  std::function<size_t()> save_bytes_cb_ = nullptr;
 
```

Comment:
why don't we just store the savestagescontroller itself? We can then use it in the future to get more stats

instead of save_cb we'll call `GetMemoryUsage() const` or smth like this

Reply:
When calling Save savestagescontroller obj is created and its save function is invoked, it can return error f.e if there is already save operaion running. So in this case we do not want to update the savestagescontroller.
The GetMemoryUsage is invoked from Info command and we dont know at which of the save stages it is invoked, it could be that not all the resources of the save are initialized. So I updated the callback only at the point when I know that the save has started and the resources are initialized.
I can instead of setting a callback from the controller at this point set the pointer to this obj in this way I can store savestagescontroller itself instead of the callback. It looks a bit odd but I can do this. What do you think?

Reply:
True, but we could check if it really started, so we can keep the pointer to the controller always assigned

Either way, let's keep the callback